.cico.pipeline

```diff
@@ -0,0 +1,200 @@
+}
+
+def notifyBuild(String buildStatus = 'STARTED') {
+    buildStatus =  buildStatus ?: 'SUCCESSFUL'
+    emailext attachmentsPattern: 'odo/*.log', body: "See: ${env.BUILD_URL}", subject: "[${env.JOB_NAME}] Build #${env.BUILD_NUMBER}: ${buildStatus}", to: 'sgk@redhat.com'
+}
+
+node('odo') {
+
+            stage('Allocate Node'){
+                env.CICO_API_KEY = readFile("${env.HOME}/duffy.key").trim()
+                duffy_rtn=sh(
+                    script: 'cico --debug node get -f value -c hostname -c comment',
+                    returnStdout: true
+                    ).trim().tokenize(' ')
+                env.DUFFY_NODE=duffy_rtn[0]
+                env.SSID=duffy_rtn[1]
+                env.CLONE_DIR = 'go/src/github.com/redhat-developer/odo'
+            }
+    
+            try{
+                stage('Pre Setup Node'){
+                    onmyduffynode '''yum -y install git wget gcc docker &&
+                    wget https://dl.google.com/go/go1.11.2.linux-amd64.tar.gz &&
+                    tar -C /usr/local -xzf go1.11.2.linux-amd64.tar.gz'''
+                }
+    
+                stage('git clone Tests') {
+                    onmyduffynode '''git clone --single-branch --branch master --depth 1 "https://github.com/redhat-developer/odo.git" ${CLONE_DIR} && 
+                    cd  ${CLONE_DIR} && ls -la'''
+                }
+    
+                stage('Build binary') {
+                        onmyduffynode '''export PATH=$PATH:/usr/local/go/bin:/root/go/bin &&
+                        go version &&
+                        cd ${CLONE_DIR} && 
+                        make goget-tools &&
+                        make cross &&
+                        tar -cvzf odo-linux-amd64.tar.gz -C dist/bin/linux-amd64 . &&
+                        tar -cvzf odo-linux-arm.tar.gz -C dist/bin/linux-arm . &&
+                        tar -cvzf odo-darwin-amd64.tar.gz -C dist/bin/darwin-amd64 . &&
+                        tar -cvzf odo-windows-amd64.tar.gz -C dist/bin/windows-amd64 . &&
+                        ls -la
+                        '''
+                }
+    
+                stage('Unit Test') {
+                        onmyduffynode '''export PATH=$PATH:/usr/local/go/bin &&
+                        cd ${CLONE_DIR} &&
+                        make test |& tee unit_test.log'''
+                }
+
+                timeout(20) {
+                    stage('Main e2e Test') {
+                        onmyduffynode '''export PATH=$PATH:/usr/local/go/bin &&
+                        cd ${CLONE_DIR} &&
+                        make &&
+                        cp odo /usr/local/go/bin &&
+                        git branch && 
+                        rpm --query centos-release &&
+                        echo "----------starting-cluster-provision-----------" &&
+                        ./scripts/minishift-setup.sh &&
```

Comment:
Wouldn't it be better to have to use the same OpenShift instance for all the test?
Right now we are creating and deleting cluster for each stage, and it takes a lot of time.

Reply:
done!